ni_measurementlink_service/_internal/parameter/serialization_strategy.py

```diff
@@ -1,118 +1,133 @@
-DoubleArrayEncoder = _vector_encoder(encoder.DoubleEncoder)
-IntArrayEncoder = _vector_encoder(encoder.Int32Encoder)
-UIntArrayEncoder = _vector_encoder(encoder.UInt32Encoder)
+FloatArrayEncoder = _vector_encoder(cast(EncoderConstructor, encoder.FloatEncoder))
+DoubleArrayEncoder = _vector_encoder(cast(EncoderConstructor, encoder.DoubleEncoder))
+IntArrayEncoder = _vector_encoder(cast(EncoderConstructor, encoder.Int32Encoder))
+UIntArrayEncoder = _vector_encoder(cast(EncoderConstructor, encoder.UInt32Encoder))
 BoolArrayEncoder = _vector_encoder(encoder.BoolEncoder)
 StringArrayEncoder = _vector_encoder(encoder.StringEncoder, is_packed=False)
 
 
-FloatDecoder = _scalar_decoder(decoder.FloatDecoder)
-DoubleDecoder = _scalar_decoder(decoder.DoubleDecoder)
-Int32Decoder = _scalar_decoder(decoder.Int32Decoder)
-UInt32Decoder = _scalar_decoder(decoder.UInt32Decoder)
-Int64Decoder = _scalar_decoder(decoder.Int64Decoder)
-UInt64Decoder = _scalar_decoder(decoder.UInt64Decoder)
-BoolDecoder = _scalar_decoder(decoder.BoolDecoder)
-StringDecoder = _scalar_decoder(decoder.StringDecoder)
+FloatDecoder = _scalar_decoder(cast(DecoderConstructor, decoder.FloatDecoder))
+DoubleDecoder = _scalar_decoder(cast(DecoderConstructor, decoder.DoubleDecoder))
+Int32Decoder = _scalar_decoder(cast(DecoderConstructor, decoder.Int32Decoder))
+UInt32Decoder = _scalar_decoder(cast(DecoderConstructor, decoder.UInt32Decoder))
+Int64Decoder = _scalar_decoder(cast(DecoderConstructor, decoder.Int64Decoder))
+UInt64Decoder = _scalar_decoder(cast(DecoderConstructor, decoder.UInt64Decoder))
+BoolDecoder = _scalar_decoder(cast(DecoderConstructor, decoder.BoolDecoder))
+StringDecoder = _scalar_decoder(cast(DecoderConstructor, decoder.StringDecoder))
 
-FloatArrayDecoder = _vector_decoder(decoder.FloatDecoder)
-DoubleArrayDecoder = _vector_decoder(decoder.DoubleDecoder)
-Int32ArrayDecoder = _vector_decoder(decoder.Int32Decoder)
-UInt32ArrayDecoder = _vector_decoder(decoder.UInt32Decoder)
-Int64ArrayDecoder = _vector_decoder(decoder.Int64Decoder)
-UInt64ArrayDecoder = _vector_decoder(decoder.UInt64Decoder)
-BoolArrayDecoder = _vector_decoder(decoder.BoolDecoder)
-StringArrayDecoder = _vector_decoder(decoder.StringDecoder, is_packed=False)
+FloatArrayDecoder = _vector_decoder(cast(DecoderConstructor, decoder.FloatDecoder))
+DoubleArrayDecoder = _vector_decoder(cast(DecoderConstructor, decoder.DoubleDecoder))
+Int32ArrayDecoder = _vector_decoder(cast(DecoderConstructor, decoder.Int32Decoder))
+UInt32ArrayDecoder = _vector_decoder(cast(DecoderConstructor, decoder.UInt32Decoder))
+Int64ArrayDecoder = _vector_decoder(cast(DecoderConstructor, decoder.Int64Decoder))
+UInt64ArrayDecoder = _vector_decoder(cast(DecoderConstructor, decoder.UInt64Decoder))
+BoolArrayDecoder = _vector_decoder(cast(DecoderConstructor, decoder.BoolDecoder))
+StringArrayDecoder = _vector_decoder(
+    cast(DecoderConstructor, decoder.StringDecoder), is_packed=False
```

Comment:
BoolDecoder *is* typed as _Decoder
https://github.com/python/typeshed/blob/main/stubs/protobuf/google/protobuf/internal/decoder.pyi#L25

But yeah the StringDecoder cast is because of the str vs memoryview issue.